SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc

```diff
@@ -439,24 +440,26 @@ void HcalDigitizer::accumulateCaloHits(edm::Handle<std::vector<PCaloHit> > const
     std::vector<PCaloHit> hcalHits;
     hcalHits.reserve(hcalHitsOrig.size());
 
-    for ( unsigned int i=0; i< hcalHitsOrig.size(); i++) {
-      DetId id(hcalHitsOrig[i].id());
-      HcalDetId hid(id);
-      
-      //      if ( id.subdetId()==2 && hid.ietaAbs()<16 ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      if ( !htopoP->validHcal(hid) ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      else
-	hcalHits.push_back(hcalHitsOrig[i]);
-    }
     //evaluate darkening before relabeling
     if(m_HEDarkening || m_HFRecalibration){
-      darkening(hcalHits);
+      darkening(hcalHitsOrig);
```

Comment:
The darkening() function expects an HcalTestNumbering ID:
https://github.com/lgray/cmssw/blob/hgc_fasttime_fromtdconset_76X/SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc#L717-L722

I'm not sure what HcalTestNumbering::unpackHcalIndex() will do with an invalid ID, but again, darkening will only be enabled in the appropriate situations. It is not a commonly used feature.